File: count-substrings-with-only-one-distinct-letter.kt
class Solution {
  fun countLetters(S: String): Int {
    var count = 0
    var sameCharLength = 0
    for (i in 0 until S.length) {
      if (i > 0 && S[i] != S[i - 1]) {
        count++
        sameCharLength = 1
      } else {
        sameCharLength++
        count += sameCharLength
      }
    }
    return count
  }
}

File: count-substrings-with-only-one-distinct-letter.py
class Solution:
  def countLetters(self, S: str) -> int:
    return reduce(
      lambda x, y: x + (y * (y + 1)) // 2,
      [len(x) for x in [list(g) for k, g in groupby(S)]],
      0,
    )
